ethdb/privateapi/ethbackend.go

```diff
@@ -270,6 +289,80 @@ func (s *EthBackendServer) EngineGetPayloadV1(ctx context.Context, req *remote.E
 	return nil, fmt.Errorf("unknown payload")
 }
 
+// EngineGetPayloadV1, retrieves previously assembled payload (Validators only)
+func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *remote.EngineForkChoiceUpdatedRequest) (*remote.EngineForkChoiceUpdatedReply, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("not a proof-of-stake chain")
+	}
+	if s.assembledBlock.Load() == nil {
+		return nil, fmt.Errorf("mining has not been enabled yet")
+	}
+	// Check if parent equate to the head
+	parent := gointerfaces.ConvertH256ToHash(req.Forkchoice.HeadBlockHash)
+	tx, err := s.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	block := s.assembledBlock.Load().(*types.Block)
+
+	if parent != rawdb.ReadHeadBlockHash(tx) || block.Header().ParentHash != parent || atomic.LoadUint32(s.waitingForBeaconChain) == 0 {
+		return &remote.EngineForkChoiceUpdatedReply{
+			Status: string(Syncing),
+		}, nil
+	}
+	var baseFeeReply *types2.H256
+	if block.Header().BaseFee != nil {
+		var baseFee uint256.Int
+		baseFee.SetFromBig(block.Header().BaseFee)
+		baseFeeReply = gointerfaces.ConvertUint256IntToH256(&baseFee)
+	}
+	var encodedTransactions [][]byte
+	buf := bytes.NewBuffer(nil)
+
+	for _, tx := range block.Transactions() {
+		buf.Reset()
+
+		err := rlp.Encode(buf, tx)
+		if err != nil {
+			return nil, fmt.Errorf("broken tx rlp: %w", err)
+		}
+		encodedTransactions = append(encodedTransactions, common.CopyBytes(buf.Bytes()))
+	}
+	// Compute the correct block hash, by setting up the right header
+	header := block.Header()
+	header.Difficulty = serenity.SerenityDifficulty
```

Comment:
stage_mining_create_block calls engine.Prepare, which in serenity consensus engine sets difficulty & nonce, so we can omit them here.